hack/scripts/sync-prow.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Check if git-filter-repo tool is installed
+if which git-filter-repo > /dev/null 2>&1; then
+  echo "git-filter-repo is installed"
+else
+  echo "please install the git-filter-repo tool per https://github.com/newren/git-filter-repo"
+  exit 1
+fi
+
+old_prow="$PWD/test-infra"
+new_prow="$PWD/prow"
+
+# Clone the old prow repo.
+if [[ ! -d "$old_prow" ]]; then
+    git clone https://github.com/kubernetes/test-infra.git
+fi
+
+# Strip down the master branch of $old_prow to only have commits that touch the the /prow directory.
+pushd "$old_prow"
+git-filter-repo --path prow
```

Comment:
How do we expand this to include the hack/ and maybe other directories needed to build and test prow? It would be good to sync those at the same time to ensure consistency between code and tools.

Reply:
Currently the stuff in `/hack` do not tend to change frequently. We can address this when needed in the future.

Reply:
If there are no changes, it's a NOP. The point is that when it does change, it's likely quite important to propagate at the same time. The sort of skew that could otherwise arise may be very difficult to troubleshoot.

Reply:
We can follow up on this issue outside this PR.